src/annex/OnDemandRequest.cpp

```diff
@@ -170,29 +170,31 @@ OnDemandRequest::operator() () {
 		std::string user_data, user_data_file;
 		std::string availability_zone, vpc_subnet, vpc_id;
 		std::string block_device_mapping, iam_profile_name;
-		StringList group_names, group_ids( securityGroupIDs.c_str() ), parameters_and_values;
+		std::vector<std::string> group_names, group_ids, parameters_and_values;
+
+		group_ids = split(securityGroupIDs);
```

Comment:
Not to be too picky, but it is more efficient (and I think clearer) ((and promotes more functional-style programming)) to intialize in one go, e.g.

```
std::vector<std::string> group_ids = split(securityGroupIDs);
```
rather than default-initialize and assign, e.g.
```
std::vector<std::string> group_ids;
group_ids = split(securityGroupIDs);
```

Reply:
Done.